File: app/src/main/java/com/pedrosoares/cotacaoapp/presentation/CurrencyContract.java
package com.pedrosoares.cotacaoapp.presentation;

import com.pedrosoares.cotacaoapp.core.base.BaseContract;
import com.pedrosoares.cotacaoapp.model.domain.CurrencyDomain;

public interface CurrencyContract {

    interface CurrencyListView {
        void populateCurrency(CurrencyDomain currencyDomain);
        void success();
        void loading();
        void error();
    }

    interface CurrencyListPresenter extends BaseContract.Presenter {
        void fetchCurrency();
    }

    interface ListenerLayout{
        boolean verifyLayout();
    }
}

File: app/src/main/java/com/pedrosoares/cotacaoapp/presentation/CurrencyContract.kt
package com.pedrosoares.cotacaoapp.presentation

import com.pedrosoares.cotacaoapp.core.base.BaseContract
import com.pedrosoares.cotacaoapp.model.domain.CurrencyDomain

interface CurrencyContract {

    interface CurrencyListView {
        fun populateCurrency(currencyDomain: CurrencyDomain)
        fun success()
        fun loading()
        fun error()
    }

    interface CurrencyListPresenter : BaseContract.Presenter {
        fun fetchCurrency()
    }

}
